### PR TITLE
Add Subject description into API.yaml

### PR DESCRIPTION
**Pull request by VeronikaRevjakina**
#309

- [x]  Subject model

- [x] Methods

- [x] Methods with id

### CODE REVIEW

**Comment by VeronikaRevjakina** [comment]
@malinink check this PR please

File: src/main/resources/api/api.yaml [hunk 1]
```diff
@@ -197,6 +197,166 @@ paths:
...
+          in: query
+          description: Id of specific user
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '200':
+          description: An array of subjects
+          schema:
+            type: array
```
Comment (malinink) [inline src/main/resources/api/api.yaml:217]: there is no need of defining array here, just use $ref V1Subjects

File: src/main/resources/api/api.yaml [hunk 2]
```diff
@@ -197,6 +197,166 @@ paths:
...
+            items:
+              $ref: '#/definitions/V1Subjects'
+        '400':
+          description: Invalid user ID supplied
```
Comment (malinink) [inline src/main/resources/api/api.yaml:221]: "Invalid query parameter value"
We could change query parameters

File: src/main/resources/api/api.yaml [hunk 3]
```diff
@@ -197,6 +197,166 @@ paths:
...
+          schema:
+              $ref: '#/definitions/V1Error'
+        '403':
+          description: Permission denied
+          schema:
+              $ref: '#/definitions/V1Error'
+        '404':
```
Comment (malinink) [inline src/main/resources/api/api.yaml:228]: 404 is impossible for list, if no subjects founded - there will be empty array.

File: src/main/resources/api/api.yaml [hunk 4]
```diff
@@ -197,6 +197,166 @@ paths:
...
+        - name: subject
+          in: body
+          required: true
+          description: Subject that needs to be created
+          schema:
+            $ref: '#/definitions/V1Subject'
+      tags:
+          - subjects
+      responses:
+        '200':
```
Comment (malinink) [inline src/main/resources/api/api.yaml:249]: use 201 for created model

File: src/main/resources/api/api.yaml [hunk 5]
```diff
@@ -197,6 +197,166 @@ paths:
...
+          description: Return a new subject
```
Comment (malinink) [inline src/main/resources/api/api.yaml:250]: nope

File: src/main/resources/api/api.yaml [hunk 6]
```diff
@@ -197,6 +197,166 @@ paths:
...
+          schema:
+            $ref: '#/definitions/V1Subject'
```
Comment (malinink) [inline src/main/resources/api/api.yaml:252]: V1Identity

File: src/main/resources/api/api.yaml [hunk 7]
```diff
@@ -197,6 +197,166 @@ paths:
...
+      parameters:
+        - name: subject
+          in: body
+          required: true
+          description: Subject that needs to be updated
+          schema:
+            $ref: '#/definitions/V1Subject'
+      responses:
+        '200':
+          description: Return the updated question
```
Comment (malinink) [inline src/main/resources/api/api.yaml:279]: empty body

File: src/main/resources/api/api.yaml [hunk 8]
```diff
@@ -197,6 +197,166 @@ paths:
...
+          - subjects
+        parameters:
+          - name: subjectId
+            in: path
+            required: true
+            description: The id of the subject to retrieve
+            type: integer
+        responses:
+          '200':
+            description: Expected response to a valid request
```
Comment (malinink) [inline src/main/resources/api/api.yaml:312]: modify descriptions from users API

File: src/main/resources/api/api.yaml [hunk 9]
```diff
@@ -297,6 +457,34 @@ definitions:
...
+      id:
+        type: integer
+        format: int64
+      name:
+        type: string
+      description:
+        type: string
+      user:
+        type: object
+        items:
```
Comment (malinink) [inline src/main/resources/api/api.yaml:475]: items???

File: src/main/resources/api/api.yaml [hunk 10]
```diff
@@ -297,6 +457,34 @@ definitions:
...
+          $ref: '#/definitions/V1Identity'
+
+    example:
```
Comment (malinink) [inline src/main/resources/api/api.yaml:478]: fix example to be correct

**Review by malinink — CHANGES_REQUESTED** [review]
Check all descriptions against user API please

**Pull request closed without merging**